homeassistant/loader.py

```diff
@@ -60,6 +60,15 @@
 MOVED_ZEROCONF_PROPS = ("macaddress", "model", "manufacturer")
 
 
+class DHCPMatcher(TypedDict):
+    """Matcher for the dhcp integration."""
+
+    domain: str
+    macaddress: str | None
```

Comment:
Can the item be present and None or can it just be missing from the dict?

Reply:
Just missing

Reply:
Then we should remove the optional typing and set `total` to `False` instead.

Reply:
We use create a typed dict with two different parents if we need to set some keys as required, ie where one parent has total set to `True`.

Reply:
06e0a0c641